src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -1145,15 +1155,34 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
     };
 
     renderActivitySidebarFilter = () => {
-        const { features } = this.props;
+        const { features, hasReplies, hasTasks } = this.props;
         const { feedItemsStatusFilter } = this.state;
-        const shouldShowActivityFeedFilter = isFeatureEnabled(features, 'activityFeed.filter.enabled');
+        const activityFeedFilterEnabled = isFeatureEnabled(features, 'activityFeed.filter.enabled');
+        const newThreadedRepliesEnabled = isFeatureEnabled(features, 'activityFeed.newThreadedReplies.enabled');
+        const shouldShowActivityFeedFilter =
+            activityFeedFilterEnabled || (newThreadedRepliesEnabled && (hasReplies || hasTasks));
 
         if (!shouldShowActivityFeedFilter) {
             return null;
         }
+
+        const activityFilterOptions: ActivityFilterOption[] = [ACTIVITY_FILTER_OPTION_ALL];
+        if (newThreadedRepliesEnabled) {
+            // Determine which filter options to show based on what activity types are available in current context
+            if (hasReplies) {
```

Comment:
This was added based on the idea that hasReplies also determines whether or not comments have the ability to be unresolved/resolved. However, after discussing it further with Brian, if the feature flip newThreadedReplies is enabled, it will always have unresolved/resolved comments. Code will be updated.